### ISSUE TITLE
Casting incomparable types should raise an error instead of a warning

### ISSUE THREAD

**Issue by piyush-J** [eisop/checker-framework#155]
```
public class Cast {
    void foo() {
        @H1S1 Object p;
        p = (@H1S1 Object) new @H1S2 Object();
    }
}
```

Output: warning: (cast.unsafe)

Expected: error: (cast.incomparable)
